process/block/shardblock.go

```diff
@@ -1152,6 +1152,7 @@ func (sp *shardProcessor) updateState(headers []data.HeaderHandler, currentHeade
 		)
 
 		sp.setFinalizedHeaderHashInIndexer(header.GetPrevHash())
+		sp.blockChain.SetHighestFinalBlockCoordinates(prevHeader.GetNonce(), prevHeaderHash, prevHeaderRootHashForPruning)
```

Comment:
`prevHeaderRootHashForPruning` is the root hash from which the previous to final block started processing. The roothash that was committed after previous to final block was processed should be `scheduledPrevHeaderRootHash`

Reply:
if we want to set the last finalized Header data then header.GetNonce, headerHash, and scheduledHeaderRootHash

Reply:
Fixed.